storage/statedb/cache_redis_test.go

```diff
@@ -31,33 +32,77 @@ var testHosts = []string{"localhost:6379"}
 
 // TODO-Klaytn: Enable tests when redis is prepared on CI
 
+func _TestSubscription(t *testing.T) {
+	channelName := "channel"
+	msg1 := "testMessage1"
+	msg2 := "testMessage2"
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	go func() {
+		cache, err := NewRedisCache(testHosts, false)
+		assert.Nil(t, err)
+
+		ch := cache.SubscriptionChannel(channelName)
+
+		actualMsg := <-ch
+		assert.Equal(t, msg1, actualMsg.Payload)
+
+		actualMsg = <-ch
+		assert.Equal(t, msg2, actualMsg.Payload)
+
+		wg.Done()
+	}()
+	time.Sleep(100 * time.Millisecond)
+
+	cache, err := NewRedisCache(testHosts, false)
+	assert.Nil(t, err)
+
+	if err := cache.Publish(channelName, msg1); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := cache.Publish(channelName, msg2); err != nil {
+		t.Fatal(err)
+	}
+
+	// to prevent continuous waiting
+	go func() {
+		time.Sleep(time.Second)
+		wg.Done()
+	}()
```

Comment:
Could you explain why this part of code is needed?

Reply:
If there are errors or unexpected problems in the code such as the redis server is not working, the test will wait to receive a message from the channel persistently. To finish this test in a short time in any circumstance, I added an additional timeout logic unlocking waitGroup.